dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -43,12 +43,19 @@ public sealed class Plan : ISKFunction
     public ContextVariables NamedParameters { get; set; } = new();
 
     /// <summary>
-    /// Named outputs for the function
+    /// Named outputs for the function -- these are used within a plan.
     /// </summary>
     [JsonPropertyName("named_outputs")]
     [JsonConverter(typeof(ContextVariablesConverter))]
     public ContextVariables NamedOutputs { get; set; } = new();
```

Comment:
I see a definite point of confusion between NamedOutputs and NamedResults. I think NamedOutputs could be changed, since it feels to me like "Plan.NamedOutputs" should be the outputs of the plan.

Maybe PlanVariables?

Reply:
Yeah @RogerBarreto and I had discussions on this last week that we need to continue. I wanted to push this change as a step in a better direction, but will pump the breaks for now.

The problem we want to solve is:
- How to identify that a plan/ISKFunction has an output that needs to be saved to the parent Plan.State
- How to identify that M<=N steps in a Plan have an output that is part of the parent output (Plan.State.ToString()).

Reply:
NamedOutputs and NamedResults are both outputs of the plan. Right now, functions have a single output. In the future that could change where a function has multiple outputs that should be preserved in the state, the main issue is knowing which (single/multiple) output is something that should be part of the default result.